packages/provider/provider/index.js

```diff
@@ -82,11 +82,11 @@ export default class Provider extends Component {
   saveState(state, fn) {
     if (!this.mounted) return debug('root component no longer mounted');
 
-    this.setState(state, () => {
+    this.setState(state, function () {
```

Comment:
pretty sure this was the intention here, though nothing is actually using the arguments here.

Reply:
Instead of a bind, you could just store `this` in a variable, and reference that instead of doing a bind call everytime saveState is called.